src/org/abymaly/code/BinaryTree.java

```diff
@@ -0,0 +1,57 @@
+package org.abymaly.code;
+
+public class BinaryTree<T> {
+
+    public class Node<T> {
+        public Integer data;
+        public Node nextLeft;
+        public Node nextRight;
+
+        public Node(Integer data) {
+            this.data = data;
+            }
+
+        public Integer getData() {
+            return data;
+        }
+
+        public Node getNextLeft() {
+            return nextLeft;
+        }
+
+        public Node getNextRight() {
+            return nextRight;
+        }
+
+        Node<T> root;
+        Node<T> actual;
+
+        public void appendNode(Integer newData) {
+            Node ref = root;
+            Node<T> newNode = new Node<>(newData);
+
+            if (root == null) {
+                root = newNode;
+                return;
+            }
+            actual = root;
+
+            while (actual != null) {
+
+                if (newData.compareTo(root.getData()) < 0) {
+                    if (actual.nextLeft == null) {
+                        nextLeft = newNode;
+                    } else {
+                        nextLeft = actual;
+                    }
+                } else if (newData.compareTo(root.getData()) > 0)
```

Comment:
Węszę podstęp. 
Może dla niektórych przypadków to działać, ale bym się nie przywiązywał do tego.
Celem istnienia pętli `while` jest (jak sądzę) wędrowanie po drzewie w celu znalezienia dogodnej miejscówki "gdzie by tu dostawić nowy węzeł".
Jeśłi tak, to niech wstawianie odbędzie się poza pętlą while. Bo istniej obawa, że nowy węzeł wstawi sie w kilku miejscach - bo pętla jeszcze raz się wykona.